README.ConfigureResourceReferences.md

```diff
@@ -0,0 +1,185 @@
+    yaml file (under either the `config/servicemappings/` folder or the
+    `scripts/resource-autogen/generated/servicemappings/`), the corresponding
+    `kind` field in the same resource config should be the KCC kind.
+
+1.  If you know the GCP resource name, you can either use the search widget to
+    find the corresponding TF type in
+    [Terraform doc](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs)
+    with the GCP resource name, then figure out the KCC kind name following the
+    step above; or you can use the search widget to find the corresponding KCC
+    kind directly in
+    [KCC reference doc](https://cloud.google.com/config-connector/docs/reference/overview).
+
+1.  If you can't find a KCC kind with the steps above, it's possible that the
+    referenced resource is not supported in KCC. Reach out to the KCC team is
+    you are unsure. If you've determined that the KCC kind is not supported in
+    KCC, skip the next couple of sections and go to
+    [Configure reference resource in the service mappings](#configure-reference-in-the-service-mappings).
+
+### Determine whether the referenced resource is TF-based or DCL-based
+
+Based on the KCC kind names of the references you figured above, follow the
+steps to identify the referenced resource is DCL or TF based.
+
+1.  Search for the KCC kind name of the referenced resource on the
+    [crds folder](https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/crds).
+
+1.  Open the CRD file containing the KCC kind name, search for `dcl2crd` and
+    `tf2crd` labels (only one should exist for a particular CRD).
+
+    1.  If `cnrm.cloud.google.com/dcl2crd: "true"` is found, then it is a
+        DCL-based resource.
+
+    1.  If `cnrm.cloud.google.com/tf2crd: "true"` is found, then it is a
+        TF-based resource.
+
+## Identify whether the reference field is a string field or a field of other types (list, etc)
+
+Check the TF schema of the target resource in the TF provider to determine the
+type of the reference field. The URL of file is in the format of
+`https://github.com/hashicorp/terraform-provider-google-beta/blob/main/google-beta/services/[servicename]/resource_[tf_type_name_without_google_prefix].go`.
+You can find a `Type` field for each argument, and two most common types are
+`schema.TypeString` and `schema.TypeList`.
+
+1.  If the field is a string, follow
+    [Configure reference resource in the service mappings](#configure-reference-in-the-service-mappings)
+    to continue configuring the resource reference.
+1.  If the field is a list:
+
+    1.  If the field is not required, mark it as an ignored field following
+        [this example](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/b746248cd5a9b30669380513de8fdc6b4c43018d/config/servicemappings/cloudbuild.yaml#L204).
+    1.  If the field is required, reach out to the KCC team for further
+        discussion.
```

Comment:
let's suggest to raise a GitHub question. For our team members we can always discuss internally.